modules/cockroachdb/options.go

```diff
@@ -1,180 +1,106 @@
+			return fmt.Errorf("unsupported user %q with TLS, use %q", user, defaultUser)
+		}
 
-// connString returns a connection string for the given host, port and options.
-func (opts options) connString(host string, port nat.Port) (string, error) {
-	cfg, err := opts.connConfig(host, port)
-	if err != nil {
-		return "", fmt.Errorf("connection config: %w", err)
+		req.Env[envUser] = user
+		return nil
 	}
-
-	return stdlib.RegisterConnConfig(cfg), nil
 }
 
-// connConfig returns a [pgx.ConnConfig] for the given host, port and options.
-func (opts options) connConfig(host string, port nat.Port) (*pgx.ConnConfig, error) {
-	user := url.User(opts.User)
-	if opts.Password != "" {
-		user = url.UserPassword(opts.User, opts.Password)
-	}
-
-	sslMode := "disable"
-	if opts.TLS != nil {
-		sslMode = "require" // We can't use "verify-full" as it might be a self signed cert.
-	}
-	params := url.Values{
-		"sslmode": []string{sslMode},
-	}
-
-	u := url.URL{
-		Scheme:   "postgres",
-		User:     user,
-		Host:     net.JoinHostPort(host, port.Port()),
-		Path:     opts.Database,
-		RawQuery: params.Encode(),
-	}
-
-	cfg, err := pgx.ParseConfig(u.String())
-	if err != nil {
-		return nil, fmt.Errorf("parse config: %w", err)
-	}
-
-	if opts.TLS != nil {
-		tlsCfg, err := opts.TLS.tlsConfig()
-		if err != nil {
-			return nil, fmt.Errorf("tls config: %w", err)
+// WithPassword sets the password when using password authentication.
+// This will be ignored if data exists in the `/cockroach/cockroach-data` directory within the container.
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.Env[envPassword] = password
+		if password != "" {
+			req.Cmd = append(req.Cmd, "--accept-sql-without-tls")
 		}
 
-		cfg.TLSConfig = tlsCfg
-	}
-
-	return cfg, nil
-}
-
-func defaultOptions() options {
-	return options{
-		User:       defaultUser,
-		Password:   defaultPassword,
-		Database:   defaultDatabase,
-		StoreSize:  defaultStoreSize,
-		Statements: DefaultStatements,
+		return nil
 	}
 }
 
-// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
-var _ testcontainers.ContainerCustomizer = (*Option)(nil)
-
-// Option is an option for the CockroachDB container.
-type Option func(*options)
-
-// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
-func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
-	// NOOP to satisfy interface.
-	return nil
-}
-
-// WithDatabase sets the name of the database to use.
-func WithDatabase(database string) Option {
-	return func(o *options) {
-		o.Database = database
-	}
-}
+// WithStoreSize sets the amount of available [in-memory storage].
+//
+// [in-memory storage]: https://www.cockroachlabs.com/docs/stable/cockroach-start#store
+func WithStoreSize(size string) testcontainers.CustomizeRequestOption {
```

Comment:
docs: let's update the docs/modules/cockroachdb.md file with the new options

Reply:
I thought I got them all, did I miss some?

Reply:
For some reason, I missed them in the review. I was probably looking at one commit, not the entire change set. My fault